daemon/cluster/services.go

```diff
@@ -134,6 +137,29 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string, queryRe
 
 		switch serviceSpec.Task.Runtime.(type) {
 		// handle other runtimes here
+		case *swarmapi.TaskSpec_Generic:
+			switch serviceSpec.Task.GetGeneric().Kind {
+			case string(types.RuntimePlugin):
+				if s.TaskTemplate.PluginSpec == nil {
+					return errors.New("plugin spec must be set")
+				}
+				b, err := proto.Marshal(s.TaskTemplate.PluginSpec)
```

Comment:
I'm curious why this isn't part of `ServiceSpecToGRPC`. I feel like needing to do this part separately, after calling `ServiceSpecToGRPC`, may be error-prone.